.github/workflows/ci.yml

```diff
@@ -297,7 +300,17 @@ jobs:
           BUILD_ARGS: "--load"
 
       - name: Run E2E Tests
-        run: make e2e E2E_TEST_FLAGS="-test.v -labels area=${{ matrix.area }}"
+        run: make e2e E2E_TEST_FLAGS="-test.v -labels area=${{ matrix.area }} --destroy-kind-cluster=false" || true
+
+      - name: show logs
+        run: |
+          kubectl logs -l app=crossplane --all-containers=true -n crossplane-system
+          kubectl get deployments -o yaml -n crossplane-system
+          echo PODS
+          kubectl get pods -o yaml -n crossplane-system
+          kubectl get configmap -n crossplane-system
+          helm get values crossplane  -n crossplane-system
+
```

Comment:
There is a function to export the logs of the kind cluster here, maybe we could export them on failures before destroying the cluster: https://github.com/kubernetes-sigs/e2e-framework/blob/main/pkg/envfuncs/kind_funcs.go

Reply:
The added tests is failing at the moment at Github, and I added this just to be able to debug/see what is going on. I am going to remove it once the test starts to pass, and we can focus on logs gathering in an separate issue.

Reply:
btw, found why the test is failing - crossplane cannot be upgraded because of resource limitations: https://github.com/crossplane/crossplane/actions/runs/5430979412/jobs/9877067067?pr=4261#step:14:729

```
0/1 nodes are available: 1 Insufficient cpu. preemption: 0/1 nodes
        are available: 1 No preemption victims found for incoming pod..'
```

Need to see how to overcome this.

Reply:
We are upgrading crossplane successfully in the composition validation test, maybe have a look at that, couldn't spot a difference, but if that's the case you should be able to reproduce it locally